coreui NoteBox component

dmfalke opened the pull request:
This PR introduces a `NoteBox` component. It is meant to be an alternative to `Banner`, integrating better on content-heavy pages, like record pages.

TODO: Add storybook entries.

asizemore commented on packages/libs/coreui/src/stories/containers/NoteBox.stories.tsx:
```diff
@@ -0,0 +1,74 @@
+
+const Template: Story<Props> = function Template(props) {
+  return (
+    <UIThemeProvider
```
Is there a reason to add the theme here? Or is it just to check that we get the correct colors regardless of theme?

asizemore commented on packages/libs/coreui/src/components/containers/NoteBox.tsx:
```diff
@@ -0,0 +1,47 @@
+  children: ReactNode;
+}
+
+const baseCss = css`
```
Perhaps additionally a default font? I believe coreui generally uses Inter or Roboto

dmfalke replied:
For content, I think we should use the same font as the rest of the site, but I will take a look at using one of these other fonts.

dmfalke replied:
I think I'm going to leave the font as-is.

asizemore commented on packages/libs/coreui/src/stories/containers/NoteBox.stories.tsx:
```diff
@@ -0,0 +1,74 @@
+      </div>
+    ),
+  },
+});
```
A story using a different type of child (for example, an image or some other non-text element) would be helpful if time allows

asizemore replied:
or an expandable one

dmfalke replied:
Yes, I agree! I will add a couple more, soon.

asizemore commented on packages/libs/coreui/src/components/containers/NoteBox.tsx:
```diff
@@ -0,0 +1,47 @@
+      : props.type === 'error'
+      ? errorCss
+      : infoCss;
+  return <div css={finalCss}>{props.children}</div>;
```
Perhaps for later, but it'd be nice to pair the icon with the NoteBox type. We picked one icon for error, warning, and info ( a looong time ago and i think it was never actually impllemented). Of course, one could pass that as part of the child, but if we want to keep coreui very opinionated then there could be a `useIcon` prop and if that is true. the appropriate icon for that NoteBox style appears.

dmfalke replied:
Agreed!

asizemore commented on packages/sites/ortho-site/webapp/wdkCustomization/js/client/records/Sequences.tsx:
```diff
@@ -662,23 +663,7 @@ export function RecordTable_Sequences(
-          {warningText}
-        </div>
-      )}
+      {warningText && <NoteBox type="error">{warningText}</NoteBox>}
```
we should either change `warningText` to `errorText` or use the `type='warning'`. Having "warning" text in an error Notebox is confusing.

dmfalke replied:
This is one of those things where someone wanted the box to be red, even though it isn't strictly an error. I will change it to `errorText` for clarity.

dmfalke replied:
Done!

asizemore reviewed (commented):
Mostly just suggestions and questions. Otherwise looks really good! And it's 1000x better than what we have.

One big one - I'm not seeing the "Read more"/"Read less" button here, but I believe i should be?

<img width="1154" alt="Screen Shot 2024-11-20 at 5 46 27 PM" src="https://github.com/user-attachments/assets/15ed341c-259e-4665-b793-b85811b4bc37">

asizemore commented:
ahh i found it. 
<img width="1227" alt="Screen Shot 2024-11-20 at 5 49 27 PM" src="https://github.com/user-attachments/assets/91de6092-8f58-483e-be8a-d1f60f59c9b4">

It's tough when so much is populated from whatever we're given from the backend!

dmfalke commented:
> Mostly just suggestions and questions. Otherwise looks really good! And it's 1000x better than what we have.
> 
> 
> 
> One big one - I'm not seeing the "Read more"/"Read less" button here, but I believe i should be?
> 
> 
> 
> <img width="1154" alt="Screen Shot 2024-11-20 at 5 46 27 PM" src="https://github.com/user-attachments/assets/15ed341c-259e-4665-b793-b85811b4bc37">
> 
> 

I fixed a bug related to this. Maybe on this branch? Anyway, I'll take a look.

dmfalke commented:
> > Mostly just suggestions and questions. Otherwise looks really good! And it's 1000x better than what we have.
> > 
> > 
> > 
> > One big one - I'm not seeing the "Read more"/"Read less" button here, but I believe i should be?
> > 
> > 
> > 
> > <img width="1154" alt="Screen Shot 2024-11-20 at 5 46 27 PM" src="https://github.com/user-attachments/assets/15ed341c-259e-4665-b793-b85811b4bc37">
> > 
> > 
> 
> I fixed a bug related to this. Maybe on this branch? Anyway, I'll take a look. 

Oh, I know what's going on here. This specific case is using a `<details>` element. It is likely closed by default, which means the overflow detection isn't triggered. This is actually kind of what we want, otherwise there would be two methods to collapse the content.

Generally, I prefer the `<details>` method, but I added the other one for an ortho use case. I probably should have argued my case 🙂

dmfalke commented:
While testing this PR, I noticed a bug that was introduced by #1270:

The persistence of record page table state was also moved to wdk-client. There is at least [one place](https://github.com/VEuPathDB/web-monorepo/blob/7b84bdc7fadf994b1fe9c16632c60f9214125ea2/packages/sites/genomics-site/webapp/wdkCustomization/js/client/components/records/GeneRecordClasses.GeneRecordClass.jsx#L1643-L1650) that uses the old `updateTableState` action, which needs to be updated to use the new action. Additionally, the [logic that handles scrolling to the selected table row](https://github.com/VEuPathDB/web-monorepo/blob/7b84bdc7fadf994b1fe9c16632c60f9214125ea2/packages/sites/genomics-site/webapp/wdkCustomization/js/client/components/common/RecordTableContainer.jsx#L60-L98) also needs to be moved to RecordTableSection.

I will make a new issue for this.

dmfalke commented:
I've added icon support, and made some additional stories to highlight the option. I also modified the padding and border width to compensate for the additional space needed for the icon.

asizemore reviewed (commented):
<img width="298" alt="Screen Shot 2024-11-26 at 8 03 51 AM" src="https://github.com/user-attachments/assets/b9e91713-8b82-4e2f-a78c-7781f3268050">

I was hoping for a good way to align the icon with the text. First i tried the `flex` + `align-items-center` route, but that only worked until the text was multiple lines. Next I tried the less elegant way of just adding a negative top margin to the icon. That worked, but requires all of the icons to be the same size. I believe that to be true, but am not 100% certain. What do you think?
